transformer_engine/pytorch/ops/basic/dropout.py

```diff
@@ -38,21 +40,39 @@ def op_forward(
     ) -> torch.Tensor:
 
         # Compute dropout if training
-        out = input_
         is_training = self.training
+        out = input_
         mask = None
         if is_training:
-            keep_prob = 1 - self.dropout_probability
-            mask = torch.empty_like(input_)
-            mask.bernoulli_(keep_prob)
-            mask *= 1 / keep_prob
-            out = out * mask
-
+            if input_.numel() % 16 != 0:
+                assert not isinstance(
+                    input_, Float8TensorBase
+                ), "fp8 dropout does not support non-16-element aligned inputs"
+                keep_prob = 1 - self.dropout_probability
+                mask = torch.empty_like(input_)
+                mask.bernoulli_(keep_prob)
+                mask *= 1 / keep_prob
+                out = out * mask
+            else:
+                if isinstance(input_, Float8TensorBase):
+                    assert self.otype is not None, "otype is not set for fp8 dropout"
+                    assert self.otype in [torch.half, torch.bfloat16], (
+                        "fp8 dropout only supports half and bfloat16 output types but got %s"
+                        % self.otype
+                    )
+                    out = torch.empty_like(input_, dtype=self.otype)
+                    _, mask = tex.dropout_fwd_fp8(input_, out, self.dropout_probability)
+                else:
+                    out, mask = tex.dropout_fwd(input_, self.dropout_probability, is_training)
+        else:
+            assert not isinstance(
+                input_, Float8TensorBase
+            ), "fp8 dropout does not support non-training mode"
```

Comment:
@timmoon10 I don't think that failing in inference mode is valid. I think if fp8 dropout cannot support inference, we should dequantize